File: 7-reverse-integer/reverse-integer.js

const reverse = (n)=> {
    
    let rev =0 ;
    let nCopy = n;
    n=Math.abs(n);
    while(n>0) {
       let rem = n%10;
        rev = (10*rev)+rem;
        n=Math.floor(n/10);
    }
    if(rev < -(2**31) || rev> (2**31)-1) return 0;
    return (nCopy < 0)? -1*rev : rev;
}

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {

        long rev = 0;

       int n = Math.abs(x);

        while(n>0) {
            int rem = n%10;
            rev =  rev*10 + rem;
            n = (n/10);
        }
        if (rev > Integer.MAX_VALUE) return 0;
        return (x<0) ? (int)-rev : (int)rev;

        
    }
}